### PR TITLE
[FEE WARNING] "Tiered" fee warning system

### PR DESCRIPTION
**Pull request by W3stside**
# Summary

Part of #896 

Adds a tiered system for fees:
>= 10% = low tier fee
>=20% = med tier fee
>=30% = high tier fee

Right now it is absolute, it's not relative to sell size or anything.

HIGH FEE (>30%):
Screenshot 2021-09-06 at 10 07 46

MEDIUM FEE (>20%):
Screenshot 2021-09-06 at 10 07 54

LOW FEE (>10%):
Screenshot 2021-09-06 at 10 08 02

  # To Test

1. make swaps and mess with amounts
- [ ] Switching tokens resets the high fee warning CHECKBOX state (if checked on change = unchecked)
- [ ] Switching tokens updates the high fee colours/message IF NECESSARY (check the tooltip)
- [ ] Flipping tokens does the same as points 1 & 2 above

### CODE REVIEW

File: src/custom/components/HighFeeWarning/index.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  gap: 2px;
+  justify-content: center;
+  align-items: center;
+`
+
+const HighFeeWarningContainer = styled(AuxInformationContainer).attrs((props) => ({
+  ...props,
+  hideInput: true,
+}))<HighFeeContainerProps>`
+  &&&&& {
```
Comment (biocom) [inline src/custom/components/HighFeeWarning/index.tsx:32]: Not a fan

File: src/custom/components/HighFeeWarning/index.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        margin: 1px 4px 0 0;
+      }
+    }
+  }
+`
+
+const ErrorStyledInfo = styled(StyledInfo)`
+  color: ${({ theme }) => theme.infoText};
+`
+const HIGH_TIER_FEE = { level: '30', colour: '#FF7676' }
```
Comment (biocom) [inline src/custom/components/HighFeeWarning/index.tsx:73]: Could consider defining theme colors and pull from there (as we can tweak/re-use the colors).

**Comment by biocom** [comment]
Nice PR, I like the approach of the tiered warning. Some comments;
- Mobile we could consider wrapping content instead:
Screen Shot 2021-09-06 at 10 29 12
- The 'swap anyway' text/checkbox is quite small. We could consider wrapping it by default, allowing a larger checkbox/test. I can handle this otherwise if you prefer to get this PR going.

**Comment by W3stside** [comment]
> - The 'swap anyway' text/checkbox is quite small. We could consider wrapping it by default, allowing a larger checkbox/test. I can handle this otherwise if you prefer to get this PR going.

was going to suggest that, would be great if you could

**Review by elena-zh — APPROVED** [review]
Changes look good to me. 
My question is related to boarder values: we say that fee amount 'exceed' a certain amount, but we start to see the warning when fee = exact this amount. 
image

From my perspective, 'exceed' means > than 10 (or 20, or 30), but not >=10 (or 20, or 30). 
Anyways, it is a nitpick. And it will be not so easy to face this issue.

**Review by fairlighteth — APPROVED** [review]
Approve with comments.

**Comment by W3stside** [comment]
@annamsgeorge merged in but feel free to post merge comment on this one or others

**Comment by annamsgeorge** [comment]
Love it!!

**Pull request merged**